cmd/otel-allocator/main.go

```diff
@@ -69,7 +73,16 @@ func main() {
 	// creates a new discovery manager
 	discoveryManager := lbdiscovery.NewManager(log, ctx, gokitlog.NewNopLogger())
 	defer discoveryManager.Close()
-	discoveryManager.Watch(allocator.SetTargets)
+
+	if filterStrategy == "none" {
```

Comment:
Hmm maybe a switch statement here where the `default` is an error (should look up tradeoff bw switch statement and if statements)?